samples/rust/rust_print.rs

```diff
@@ -34,6 +34,28 @@ fn arc_print() -> Result {
     // Uses `dbg` to print, will move `c` (for temporary debugging purposes).
     dbg!(c);
 
+    #[cfg(CONFIG_RUST_COERCE_POINTEE)]
+    {
+        use core::fmt::Display;
+        fn arc_dyn_print(arc: &Arc<dyn Display>) {
+            pr_info!("Arc<dyn Display> says {arc}");
+        }
+        // `Arc` can be used to delegate dynamic dispatch and the following is an example.
+        // Both `i32` and `&str` implements `Display`.
+        // This enables us to express a unified behaviour, contract or protocol on both `i32` and `&str`
+        // in a single `Arc` type `Arc<dyn Display>`.
+        let a_i32_display: Arc<dyn Display> = Arc::new(42i32, GFP_KERNEL)?;
+        let a_str_display: Arc<dyn Display> = a.clone();
+        arc_dyn_print(&a_i32_display);
+        arc_dyn_print(&a_str_display);
+    }
+
+    #[cfg(not(CONFIG_RUST_COERCE_POINTEE))]
+    pr_info!(
+        "The demonstration for dynamic dispatching through Arc is skipped. "
+        "To see the print-out from this example, please upgrade your Rust compiler to 1.83.0 or higher.\n"
+    );
```

Comment:
Why is this example limited. Does the logic we have pre-1.83 not work for the sample?